src/openms_gui/include/OpenMS/VISUAL/LayerData.h

```diff
@@ -507,8 +517,71 @@ namespace OpenMS
     ExperimentType::SpectrumType cached_spectrum_;
   };
 
+  class LogWindow;
+
+  /// A base class to annotate layers of specific types with (identification) data
+  class LayerAnnotatorBase
```

Comment:
Maybe not relevant here but we should also consider an interface for online calculation of annotations while browsing through spectra.
E.g., one use case would be to let the user dynamically choose masses that should be annotated. Another one would be m/z and charge annotation of "relevant" peaks depending on the zoom level.